### DISCUSSION TITLE
[Azure Plugin] Service Principal via ENV vars

### DISCUSSION THREAD

**Discussion by dharapvj** [coredns/coredns#4611] [Q&A] [answered]
Hello,

Currently looks like Azure Plugin does not support picking Azure service principal details ENV variables. This means I must write everything include CLIENT_SECRET in CoreDNS Corefile.

I would like to avoid this. AWS Route53 plugin is supporting env variables.

Is there a way to use env vars to provide client_secret?

If not.. may I try to provide pull request?

**Comment by chrisohaver** [comment, marked as answer]
https://coredns.io/manual/configuration/#environment-variables

**Reply by dharapvj** [reply]
Thank you that works!
